WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderDiscoverViewModel.kt

```diff
@@ -86,6 +89,14 @@ class ReaderDiscoverViewModel @Inject constructor(
     private val photonWidth: Int = (displayUtilsWrapper.getDisplayPixelWidth() * PHOTON_WIDTH_QUALITY_RATION).toInt()
     private val photonHeight: Int = (photonWidth * FEATURED_IMAGE_HEIGHT_WIDTH_RATION).toInt()
 
+    private val communicationChannelObserver = Observer { data: Event<ReaderDiscoverCommunication>? ->
```

Comment:
I've moved the observer into a field so we can remove it in onCleared method.

Reply:
I like the way you moved the communicationChannel stream logic to handlers!